### ISSUE TITLE
problem sending messages with group-id set using amqp

### ISSUE THREAD

**Issue by alfintatorkace** [Azure/azure-uamqp-c#21]
I modified the message_sender and message receiver samples to try to set/send and receive/get the group_id (SessionId in service bus explorer).

If I send a message to my queue with service bus explorer while setting SessionId field, I can run message_receiver_sample, receive messages and get the SessionId by using properties_get_group_id.

If I try to use properties_set_group_id in message_sender_sample, service bus explorer shows messages in the queue but I can't receive them with either service bus explorer or message_receiver_sample. The messages just seem to get stuck in the queue forever until I delete the queue.

See attached samples (I removed my access key but left in my host/queue name to show what I did).

[message_sender_receiver.zip](https://github.com/Azure/azure-uamqp-c/files/483604/message_sender_receiver.zip)

alf

**Comment by dcristoloveanu** [comment]
Hi @alfintatorkace 

Sorry for the extremely big delay in the reply.
I just tried setting group-id and reply-group-id on messages sent to a queue and that did work (as in it shows up in SB Explorer). I simply used

                properties_set_group_id(properties, "DEADBEEF");
                properties_set_reply_to_group_id(properties, "DEADBEEF");

I noticed however that the SB Explorer listener does not pick up these messages from the queue, although I have not investigated whether the listener is supposed to treat the messages that have a group-id differently.

![image](https://cloud.githubusercontent.com/assets/8507399/21778255/86340e62-d657-11e6-90d6-df62784b4552.png)

If there is still an issue with this please re-open it and I will investigate further.

Cheers,
/Dan

**Comment by alfintatorkace** [comment]
Hi Dan,

It’s been a while, and on to other projects, but we eventually ended up using qpid_proton stuff which just worked.

The issue I was seeing was when I added these properties in SB Explorer I was able to send/receive them in SB Explorer.
When I sent them using the modified amqp samples, I could see them in the queue, but I was not able to receive them with either SB explorer or the modified sample.

Mark Radbourne later (in November) said it was all working fine for him with the latest Azure backend, and I was on to other things by then and didn’t have access to my old portal any more.  I attached the last thread I had with him.

alf
